pkg/querier/queryrange/results_cache.go

```diff
@@ -154,63 +192,97 @@ func (s resultsCache) handleHit(ctx context.Context, r *Request, extents []Exten
 
 		accumulator.TraceId = jaegerTraceID(ctx)
 		accumulator.End = extents[i].End
-		accumulator.Response, err = mergeAPIResponses([]*APIResponse{accumulator.Response, extents[i].Response})
+		accRes, err := accumulator.toResponse()
```

Comment:
Hrm, we are serializing _and_ serializing for every iteration here. Seems wasteful, can we optimize?

Reply:
I'll check if I can use different type for accumulating, so I don't loose the type.

Reply:
I've updated the code to avoid the accumulator serializing and deserializing. Let me know if that works for you.